lib/workers/global/config/parse/host-rules-from-env.ts

```diff
@@ -23,12 +66,19 @@ export function hostRulesFromEnv(env: NodeJS.ProcessEnv): HostRule[] {
     // Double underscore __ is used in place of hyphen -
     const splitEnv = envName.toLowerCase().replace(/__/g, '-').split('_');
     const hostType = splitEnv.shift()!;
-    if (datasources.has(hostType)) {
-      const suffix = splitEnv.pop()!;
-      if (isAuthField(suffix)) {
+    if (
+      datasources.has(hostType) ||
+      (platforms.has(hostType) && splitEnv.length > 1)
+    ) {
+      let suffix = splitEnv.pop()!;
+      if (isAuthField(suffix) || isHttpsAuthField(suffix)) {
+        if (isHttpsAuthField(suffix)) {
+          suffix = restoreHttpsAuthField(suffix);
+        }
```

Comment:
```suggestion
      if (isHttpsAuthField(suffix)) {
        suffix = restoreHttpsAuthField(suffix);
```

remove unnecessary checks